ui/src/sandbox.rs

```diff
@@ -167,9 +176,21 @@ impl Sandbox {
 
     pub fn execute(&self, req: &ExecuteRequest) -> Result<ExecuteResponse> {
         try!(self.write_source_code(&req.code));
-        let mut command = self.execute_command(req.channel, req.mode, req.crate_type, req.tests, req.backtrace, req.edition);
 
-        let output = try!(command.output().map_err(Error::UnableToExecuteCompiler));
+        let mut command = self.execute_command(req.channel, req.mode, req.crate_type, req.tests, req.backtrace, req.edition);
+        let output = if let Some(ref input) = req.stdin {
+            command.stdin(Stdio::piped());
+            command.stdout(Stdio::piped());
+            command.stderr(Stdio::piped());
+            let mut child = try!(command.spawn().map_err(Error::UnableToExecuteCompiler));
```

Comment:
Feel free to use `?` instead of `try!` — this code is just comparatively old and predates `?`. I'm half-planning on doing a once-over the code to try and clean things up when the 2018 edition lands in stable.

Reply:
I just wanted to keep it inline with the style used everywhere.

Reply:
It's not everywhere though 😉 Newer code, like the tests, use `?`. Anyway, use whichever you prefer and it will get cleaned up at some point.